app/static/src/app/components/code/CodeTab.vue

```diff
@@ -3,10 +3,18 @@
         <generic-help title="Write odin code" :markdown="codeHelp"></generic-help>
         <code-editor/>
         <button class="btn btn-primary mt-2" id="compile-btn" :disabled="!codeIsValid" @click="compile">Compile</button>
-        <div class="mt-2" id="code-status">
-            <vue-feather class="inline-icon" :class="iconClass" :type="validIcon"></vue-feather>
+        <div v-show="!codeValidating" class="mt-2" id="code-status">
+            <vue-feather class="inline-icon me-1"
+                         :class="iconClass"
+                         :type="validIcon"
+                         :size="20"
+                         :stroke-width="4"></vue-feather>
             {{ validMsg }}
         </div>
+        <div v-show="codeValidating" class="mt-2" id="code-loading">
+            <span class="spinner-border spinner-border-sm me-2 text-warning"></span>
```

Comment:
I think this is a bit too much - as a user it makes me want to pause between every validation, which is a bit jarring.

Can we do this more subtly - like don't change the string, and use a greyed out tick (or even also take down the grey value on the string a bit too)

Reply:
yep did it more subtly! just greys